docs/guides/c-extension-for-python.mdx

```diff
@@ -0,0 +1,226 @@
+
+As an example, assume you write a C function to build an observable and would like to return
+it to Python. You can convert a C-side `QkObs*` to a Python-side `SparseObservable` object
+(both are backed by the same Rust data structure), using the provided converter
+`qk_obs_to_python`:
+```c
+// file: extension.c
+#define PY_SSIZE_T_CLEAN
+#include <Python.h>  // include Python header for access to PyObject
+#define QISKIT_C_PYTHON_INTERFACE  // enable C->Python conversion functions
+#include <qiskit.h>
+
+PyObject *build_observable(void) {
+    QkObs *obs = qk_obs_zero(100);
+    // build the observable ...
+    PyObject *pyobj = qk_obs_to_python(obs); // convert to Qiskit's Python ``SparseObservable``
+    qk_obs_free(obs);
+    return pyobj;
+}
+```
+
+The following demonstrates how to compile this into a shared library - for example, `qiskit_cextension.so`.
+Once this is done, you can call the C program from Python:
+```python
+# file: main.py
+import qiskit
+import ctypes
+
+# Load the extension, ensuring the global interpreter lock (GIL) is acquired for function calls,
+# which you need for the C->Python object conversion.
+lib = ctypes.PyDLL("/path/to/qiskit_cextension.so")
+lib.build_observable.argtypes = None  # set argument types to the function
+lib.build_observable.restype = ctypes.py_object  # set return type
+
+# now you can directly call the function
+obs = lib.build_observable()
+print("SparseObservable instance?", isinstance(obs, qiskit.quantum_info.SparseObservable))
+print(obs)
+```
+
+## Build
+
+First, you have to build the Qiskit Python extension. This includes the C symbols so you can access both interfaces via the same shared library. This is important to ensure data can be passed
+correctly across C and Python.
+```bash
+python setup.py build_rust --inplace --release
+```
+The shared library is called ``_accelerate.<platform-specific-part>``. Find its location and name as follows:
+```bash
+QKLIB=$(python -c "import os; import qiskit; print(os.path.dirname(qiskit._accelerate.__file__))")
+QKNAME=$(python -c "import os; import qiskit; print(os.path.basename(qiskit._accelerate.__file__))")
+```
+
+You will need to know the location of the environments Python includes (``Python.h``) and libraries (``libpython.<suffix>``).
+These can, for example, be identified with
+```bash
+PYINCLUDE=$(python -c "import sysconfig; print(sysconfig.get_path('include'))")
+PYLIB=$(python -c "import sysconfig; print(sysconfig.get_config_var('LIBDIR'))")
+PYNAME=$(find $PYLIB -maxdepth 1 -name "libpython*" | grep -oE "[^/]+$" | grep -oE "python[0-9]+\.[0-9]+" || echo "python")
+```
+(If you already know these locations and names, you can also just set them directly.)
+
+### Linux
+
+Linking can differ among platforms. The following describes a solution for linkers
+supporting libraries with arbitrary names, using the `-l:` flag. See below if your linker
+requires the library to be called `lib<something>`.
```

Comment:
The colon in `-l:<pathspec>` is specific to GNU `ld`.  Using Linux isn't a sufficient or necessary condition for the GNU toolchain to be in use, it's just that Linux systems often distribute the GNU toolchain by default, and other platforms tend not to.

Reply:
So maybe we can just say to use `-l:` if the linker supports it and otherwise use the symlink solution?